### PR TITLE
[MRG]: Puts deployment into a script

### PR DESCRIPTION
**Pull request by thomasjpfan**
Closes https://github.com/dnouri/skorch/issues/353, https://github.com/dnouri/skorch/issues/129

This PR adds a `deploy.sh` script that can deploy to the staging or live pypi server:

```bash
./scripts/deploy.sh stage

./scripts/deploy.sh live
```

The script runs `pytest -x` first to make sure the test pass before deploying.

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Thanks, I will look at this on Monday.

**Comment by thomasjpfan** [comment]
@ottonemo This PR was updated to address your points. It would be good to test this deploy script out on test.pypi.org to make sure the README.rst is formatted correctly. I have been testing it here: https://test.pypi.org/project/skorch-doc-test/ and it seems fine.

Since the deploy script was moved, one needs to run, `./scripts/deploy.sh stage`, to deploy to the test.pypi.org server.

File: scripts/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+trap remove_env EXIT
+
+source activate $CONDA_ENV
+conda install -q -y "--file=${DEV_REQ}"
+conda install -q -y "twine==${TWINE_VERSION}"
+conda install -c pytorch -y "pytorch==${PYTORCH_VERSION}"
+python setup.py install
+
+# Makes sure CONDA_ENV's pytest and twine sourced
+source activate $CONDA_ENV
```
[the remark this answers is not in the document]
Comment (thomasjpfan) [reply scripts/deploy.sh]: It's not needed. This has been removed.

File: scripts/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+set -e
+
+PYTORCH_VERSION="0.4.1"
+PYTHON_VERSION="3.6"
+TWINE_VERSION="1.12.1"
+CONDA_ENV="skorch-deploy"
+CONDA_ENV_YML="environment.yml"
+DEPLOY_REQ="scripts/requirements-deploy.txt"
```
[the remark this answers is not in the document]
Comment (thomasjpfan) [reply scripts/deploy.sh]: Yup, it is a leftover. This line has been removed.

**Comment by benjamin-work** [comment]
> I have been testing it here: https://test.pypi.org/project/skorch-doc-test/ and it seems fine.

Just FYI, for me the README is broken.

**Comment by thomasjpfan** [comment]
Something in the last few commits broke the readme. I am looking into it.

**Comment by thomasjpfan** [comment]
I updated the wheel and pip versions, and it seems like it works again.

File: scripts/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+CONDA_ENV="skorch-deploy"
+CONDA_ENV_YML="environment.yml"
+DEV_REQ="requirements-dev.txt"
+
+if [[ $# -gt 1 ]] || [[ $1 != "live" && $1 != "stage" ]]; then
+	echo "Usage $0 [live|stage]" >&2
+	exit 1
+fi
+
+conda update -q conda
```
Comment (benjamin-work) [inline scripts/deploy.sh]:
```suggestion
conda update -q -y conda
```

**Comment by benjamin-work** [comment]
Thanks for this great addition.

**Pull request merged**